aea/protocols/generator.py

```diff
@@ -161,56 +168,71 @@ def _message_class_str(self) -> str:
+        cls_str += '        \"\"\"Get the performative of the message.\"\"\"\n'
+        cls_str += '        assert self.is_set(\"performative\"), \"performative is not set\"\n'
+        cls_str += '        return cast(str, self.get(\"performative\"))\n\n'
+        cls_str += '    @property\n'
+        cls_str += '    def contents(self) -> Dict:\n'
+        cls_str += '        \"\"\"Get the contents of the message.\"\"\"\n'
+        cls_str += '        assert self.is_set(\"contents\"), \"contents is not set\"\n'
+        cls_str += '        return cast(Dict, self.get(\"contents\"))\n\n'
+        all_contents = self._extract_all_contents()
+        for content_name, content_type in all_contents.items():
+            cls_str += '    @property\n'
+            cls_str += '    def {}(self) -> {}:\n'.format(content_name, content_type)
+            cls_str += '        \"\"\"Get \'{}\' from the contents of the message.\"\"\"\n'.format(content_name)
+            cls_str += '        assert self.is_set(\"{}\"), \"\\\'{}\\\' is not set\"\n'.format(content_name, content_name)
+            cls_str += '        return cast({}, self.get(\"{}\"))\n\n'.format(content_type, content_name)
+
+        # check_consistency() method
         cls_str += '    def check_consistency(self) -> bool:\n'
         cls_str += str.format('        \"\"\"Check that the message follows the {} protocol.\"\"\"\n', self.protocol_specification.name)
         cls_str += '        try:\n'
 
-        cls_str += '            assert self.is_set(\"message_id\"), \"message_id is not set\"\n'
-        cls_str += '            message_id = self.get(\"message_id\")\n'
-        cls_str += '            assert type(message_id) == int, \"message_id is not int\"\n\n'
-
-        cls_str += '            assert self.is_set(\"target\"), \"target is not set\"\n'
-        cls_str += '            target = self.get(\"target\")\n'
-        cls_str += '            assert type(target) == int, \"target is not int\"\n\n'
-
-        cls_str += '            assert self.is_set(\"performative\"), \"performative is not set\"\n'
-        cls_str += '            performative = self.get(\"performative\")\n'
-        cls_str += '            assert type(performative) == str, \"performative is not str\"\n\n'
-
-        cls_str += '            assert self.is_set(\"contents\"), \"contents is not set\"\n'
-        cls_str += '            contents = self.get(\"contents\")\n'
-        cls_str += '            assert type(contents) == dict, \"contents is not a dictionary\"\n'
-        cls_str += '            contents = cast(Dict, contents)\n\n'
+        cls_str += '            assert type(self.message_id) == int, \"message_id must be \'int\' but it is not.\"\n'
+        cls_str += '            assert type(self.target) == int, \"target must be \'int\' but it is not.\"\n'
+        cls_str += '            assert type(self.performative) == str, \"performative must be \'str\' but it is not.\"\n'
```

Comment:
In our actual implementation, we create an enum for the performatives. We should incorporate this here too!